File: leetcode/easy/206_反转链表/206_反转链表.py
# 双指针法
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        temp = None    # 保存cur的下一个节点
        cur = head
        pre = None
        while cur:
            temp = cur.next
            cur.next = pre
            pre = cur
            cur = temp
        return pre

File: leetcode/easy/206_反转链表/206_反转链表.go
//双指针法
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func reverseList(head *ListNode) *ListNode {
    var temp *ListNode   // 保存cur的下一个节点
    cur := head
    var pre *ListNode
    for cur != nil {
        temp = cur.Next
        cur.Next = pre
        pre = cur
        cur = temp
    }
    return pre
}